sqlmesh/core/table_diff.py

```diff
@@ -313,14 +328,16 @@ def name(e: exp.Expression) -> str:
                     ).as_("null_grain"),
                     *comparisons,
                 )
-                .from_(exp.alias_(self.source, "s"))
-                .join(
-                    self.target,
-                    on=self.on,
-                    join_type="FULL",
-                    join_alias="t",
-                )
-                .where(self.where)
+                .from_(exp.alias_(source_identifier, "s"))
```

Comment:
doesn’t seem right since it’s an identifier, is there no error here?

Reply:
```
>>> exp.alias_(exp.to_identifier("__source"), "s").sql()
'__source AS s'
```

It got the correct result. What should I have done?

Reply:
exp.tables(…).as_()